examples/compute-boids/main.rs

```diff
@@ -0,0 +1,391 @@
+                stride: 4 * 4,
+                step_mode: wgpu::InputStepMode::Instance,
+                attributes: &[
+                    // instance position
+                    wgpu::VertexAttributeDescriptor {
+                        offset: 0,
+                        format: wgpu::VertexFormat::Float2,
+                        shader_location: 0,
+                    },
+                    // instance velocity
+                    wgpu::VertexAttributeDescriptor {
+                        offset: 2 * 4,
+                        format: wgpu::VertexFormat::Float2,
+                        shader_location: 1,
+                    },
+                ]
+            },
+            wgpu::VertexBufferDescriptor {
+                stride: 2 * 4,
+                step_mode: wgpu::InputStepMode::Vertex,
+                attributes: &[
+                    // vertex positions
+                    wgpu::VertexAttributeDescriptor {
+                        offset: 0,
+                        format: wgpu::VertexFormat::Float2,
+                        shader_location: 2,
+                    },
+                ]
+            },
+        ],
+        sample_count: 1,
+        sample_mask: !0,
+        alpha_to_coverage_enabled: false,
+    });
+
+
+    // create compute pipeline
+
+    let compute_pipeline = device.create_compute_pipeline(&wgpu::ComputePipelineDescriptor {
+        layout: &compute_pipeline_layout,
+        compute_stage: wgpu::ProgrammableStageDescriptor {
+            module: &boids_module,
+            entry_point: "main",
+        },
+    });
+
+
+    // create swap chain
+
+    let mut swap_chain = device.create_swap_chain(
+        &surface,
+        &wgpu::SwapChainDescriptor {
+            usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+            format: wgpu::TextureFormat::Bgra8UnormSrgb,
+            width: size.width.round() as u32,
+            height: size.height.round() as u32,
+            present_mode: wgpu::PresentMode::Vsync,
+        },
+    );
+
+
+    // create depth texture attachment
+
+    let depth_texture = device.create_texture(&wgpu::TextureDescriptor {
+        size: wgpu::Extent3d { 
+            width: size.width.round() as u32,
+            height: size.height.round() as u32,
+            depth: 1,
+        },
+        mip_level_count: 1,
+        array_layer_count: 1,
+        sample_count: 1,
+        dimension: wgpu::TextureDimension::D2,
+        format: wgpu::TextureFormat::Depth24PlusStencil8,
+        usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+    });
+
+
+    // buffer for the three 2d triangle vertices of each instance
+
+    let vertex_buffer_data: Vec<f32> = [-0.01, -0.02, 0.01, -0.02, 0.00, 0.02].to_vec();
+    let vertices_buffer = device.create_buffer_with_data(vertex_buffer_data.as_bytes(), 
+        wgpu::BufferUsage::VERTEX | wgpu::BufferUsage::COPY_DST);
+
+
+    // buffer for simulation parameters uniform
+
+    let sim_param_data: Vec<f32> = [
```

Comment:
nit: you can instead do:
```rust
let sim_param_data = [
  0.04f32, // ...
  0.1, // ...
  0.025, ...
  ...
];
```
There is no need to make it a `Vec`, and there is no need for explicit `<f32>` if you can just make the first member `0.04f32`